pykrige/rk.py

```diff
@@ -444,3 +453,294 @@ def score(self, p, x, y, sample_weight=None, **kwargs):
+        """
+        Calculate the residuals.
+
+        Parameters
+        ----------
+        x: ndarray
+            ndarray of (x, y) points. Needs to be a (Ns, 2) array
+            corresponding to the lon/lat, for example.
+
+        Returns
+        -------
+        residual: ndarray
+            kriged residual values
+        """
+
+        krig_pred = [self.krige[i].predict(
+            x=x, **kwargs) for i in range(len(self.classes_) - 1)]
+
+        return np.vstack(krig_pred).T
+
+    def score(self, p, x, y, sample_weight=None, **kwargs):
+        """
+        Overloading default classification score method.
+
+        Parameters
+        ----------
+        p: ndarray
+            (Ns, d) array of predictor variables (Ns samples, d dimensions)
+            for classification
+        x: ndarray
+            ndarray of (x, y) points. Needs to be a (Ns, 2) array
+            corresponding to the lon/lat, for example.
+            array of Points, (x, y, z) pairs of shape (N, 3) for 3d kriging
+        y: ndarray
+            array of targets (Ns, )
+        """
+        return accuracy_score(
+            y_pred=self.predict(p, x, **kwargs), y_true=y, sample_weight=sample_weight
+        )
+
+
+def closure(data, k=1.0):
+    """Apply closure to data, sample-wise. 
+    Adapted from https://github.com/ofgulban/compoda.
+
+    Parameters
+    ----------
+    data : 2d numpy array, shape [n_samples, n_measurements]
+        Data to be closed to a certain constant. Do not forget to deal with
+        zeros in the data before this operation.
+    k : float, positive
+        Sum of the measurements will be equal to this number.
+
+    Returns
+    -------
+    data : 2d numpy array, shape [n_samples, n_measurements]
+        Closed data.
+
+    Reference
+    ---------
+    [1] Pawlowsky-Glahn, V., Egozcue, J. J., & Tolosana-Delgado, R.
+        (2015). Modelling and Analysis of Compositional Data, pg. 9.
+        Chichester, UK: John Wiley & Sons, Ltd.
+        DOI: 10.1002/9781119003144
+    """
+    data_sum = np.sum(data, axis=1)
+    out = np.copy(data)
+    for i in range(data.shape[1]):
+        out[:, i] = np.divide(out[:, i], data_sum)
+    out = out * k
+    return out
+
+
+def ilr_transformation(data):
+    """Isometric logratio transformation (not vectorized).
+    Adapted from https://github.com/ofgulban/compoda.
+
+    Parameters
+    ----------
+    data : 2d numpy array, shape [n_samples, n_coordinates]
+        Barycentric coordinates (closed) in simplex space.
+
+    Returns
+    -------
+    out : 2d numpy array, shape [n_samples, n_coordinates-1]
+        Coordinates in real space.
+
+    Reference
+    ---------
+    [1] Pawlowsky-Glahn, V., Egozcue, J. J., & Tolosana-Delgado, R.
+        (2015). Modelling and Analysis of Compositional Data, pg. 37.
+        Chichester, UK: John Wiley & Sons, Ltd.
+        DOI: 10.1002/9781119003144
+    """
+    data = data + np.finfo(float).eps
```

Comment:
I think ensuring all classes have positive probabilities and no ilr components become infinite ends up being important. For instance, to calculate "residuals" on line [582](https://github.com/GeoStat-Framework/PyKrige/blob/e18fbdd9a7f4cf3d287112a4a45c19b007567b64/pykrige/rk.py#L582), each hot-encoded class must have finite `ilr` components and residuals.

Reply:
Then, instead of adding `np.finfo(float).eps`, I would do somthing like:
```python
data = np.maximum(data, np.finfo(float).eps)
```
To only alter data where it is needed.

Reply:
Modified it to `data = np.maximum(data, np.finfo(float).eps)
`